lib/Doctrine/DBAL/Migrations/Version.php

```diff
@@ -111,9 +111,12 @@ public function __construct(Configuration $configuration, $version, $class)
         $this->migration = new $class($this);
         $this->version = $version;
 
-        $schemaProvider = new SchemaDiffProvider($this->connection->getSchemaManager(),
-            $this->connection->getDatabasePlatform());
-        $this->schemaProvider = new LazySchemaDiffProvider(new LazyLoadingValueHolderFactory(), $schemaProvider);
+        $this->schemaProvider = $schemaProvider;
+        if(is_null($this->schemaProvider)) {
```

Comment:
Does this 
```php
/** @var SchemaDiffProviderInterface */
    private $schemaProvider;
```
means that the property can never be null ?

It will never be null it's just the parameter of the constructor that is optional.